beacon-chain/rpc/attester_server.go

```diff
@@ -36,24 +36,29 @@ func (as *AttesterServer) AttestHead(ctx context.Context, att *pbp2p.Attestation
 		return nil, err
 	}
 
+	as.p2p.Broadcast(ctx, &pbp2p.AttestationAnnounce{
+		Hash: h[:],
+	})
+
 	// Update attestation target for RPC server to run necessary fork choice.
 	// We need to retrieve the head block to get its parent root.
-	blk, err := as.beaconDB.Block(bytesutil.ToBytes32(att.Data.BeaconBlockRootHash32))
+	head, err := as.beaconDB.Block(bytesutil.ToBytes32(att.Data.BeaconBlockRootHash32))
 	if err != nil {
 		return nil, err
 	}
+	// If the head block is nil, we can't save the attestation target.
+	if head == nil {
+		return nil, nil
```

Comment:
No error?

Reply:
Also should we broadcast the attestation if this fails?

Reply:
fair point, ill address them right now